cmake/Modules/Generate.cmake

```diff
@@ -30,6 +30,22 @@ function(IDLC_GENERATE)
     set(_idlc_depends CycloneDDS::idlc)
   endif()
 
+  # find idlcxx shared library
+  if(IDLC_CXX_SUPPORT)
```

Comment:
Thanks @mtudan, this would work but it would pull in knowledge of the C++ binding into the core repository, and we've tried our best to avoid doing that so far. Looking at the changes, it seems to me that it would be possible to avoid that by passing in two different arguments and removing the `CXX_SUPPORT` one: the backend library and a list of generated suffixes. For example something like:
```
set(one_value_keywords TARGET BACKEND)
set(multi_value_kewords FILES FEATURES INCLUDES SUFFIXES)
```
Where the defaults for the SUFFIXES would be ".c .h" and nothing for BACKEND. Actually, if I'd try it, I'd make a `IDLC_GENERATE_GENERIC` for that, and then let all language bindings use to that, including the C one. Then the C++ version could do
```
    if(CMAKE_CROSSCOMPILING)
       find_library(_idlcxx_shared_lib cycloneddsidlcxx NO_CMAKE_FIND_ROOT_PATH REQUIRED)

       if(_idlcxx_shared_lib)
         set(_idlcxx_depends "")
       else()
         message(FATAL_ERROR "Cannot find idlcxx shared library")
       endif()
     else()
       set(_idlcxx_shared_lib "$<TARGET_FILE:CycloneDDS-CXX::idlcxx>")
       set(_idlcxx_depends CycloneDDS-CXX::idlcxx)
     endif()

    IDLC_GENERATE_GENERIC(... BACKEND ${_idlcxx_shared_lib} SUFFIXES .hpp)
```

That seems to cover the bits where there is now something dependent on whether or not C++ is being targeted.

What do you think?

Reply:
@eboasson I think all of your points are valid, e.g. I wasn't aware of the effort put to avoid the knowledge of the C++ binding into the core repository. I did changes according to your suggestions, if there is still something you think it can be done better don't hesitate to contact me.